src/extensions/default/InlineColorEditor/main.js

```diff
@@ -59,7 +67,45 @@ define(function (require, exports, module) {
         } while (match && (pos.ch < start || pos.ch > end));
 
         if (!match) {
-            return null;
+            // Check if the cursorLine has a CSS rule of type color
+            var cssPropertyName, semiColonPos, colonPos, colorValue, cursorLineSubstring, firstCharacterPos;
+
+            // Get the css property name after removing spaces and ":" so that we can check for it in the file ColorProperties.json
+            cssPropertyName = cursorLine.split(':')[0].trim();
+
+            if (!cssPropertyName || !properties[cssPropertyName]) {
+                return null;
+            }
+
+            if (properties[cssPropertyName].type === "color") {
+                colonPos = cursorLine.indexOf(":");
+                semiColonPos = cursorLine.indexOf(";");
+                if (semiColonPos !== -1) {
+                    // edit the color value of an existing css rule
```

Comment:
I dont think you need this portion of your code.
Reason : If there is a match the code above will take care of it and control will never reach here anyways.
What you are doing here is assuming that if there is a semi colon at the end of your tag, there has to be a color matching the regex before it.
However that is not the case.
I may write
```css
color : ;
``` 
and press Ctrl + E after placing the cursor before the semi colon. That is the case that you need to tackle here.